### PR TITLE
PUDL v0.5.0 release candidate

### PR DESCRIPTION
**Pull request by zaneselvans**
Before this gets merged into `main`:
* [x] Remove `xfail` decorators from tests (like ID mapping) that should be passing now
* [x] Run `tox -e nuke`
* [x] Address any other tests / validation issues that arise
* [x] Review Documentation:
    * [x] Change reference to `ferc1_dbf2tbl` [Settings Files docs](https://catalystcoop-pudl.readthedocs.io/en/dev/dev/settings_files.html) to the new location and uses a `:py:const:` reference.
    * [x] Update reference to #1211 to reflect that it's been completed
    * [x] Clarify that EPA CEMS is (still) available from 1995-2020
    * [x] Search the docs for any references to `pudl.constants`, which has mostly been removed at this point.
    * [x] Search the docs for "data package" and "datapkg" to make sure we cleaned up any lingering references. 
    * [x] Review [data access instructions](https://catalystcoop-pudl.readthedocs.io/en/dev/data_access.html) for errors
    * [x] Review the [README](https://catalystcoop-pudl.readthedocs.io/en/dev/index.html) for errors.
    * [x] Have everyone review the [Release Notes](https://catalystcoop-pudl.readthedocs.io/en/dev/release_notes.html)
    * [x] Have everyone review [Annual Updates](https://catalystcoop-pudl.readthedocs.io/en/dev/dev/annual_updates.html)
* [x] Merge in all of the @dependabot dependency update PRs.
* [x] Update `eia860`/`eia923` row count expectations
* [x] Update `ferc1` row count expectations
* [x] Update `mcoe` row count expectations 
* [x] Check if `gf_eia923` is aggregating on correct primary keys
* [x] Check if `bf_eia923` is aggregating on correct primary keys
* [x] Check if `gfn_eia923` is aggregating on correct primary keys
* [x] Check if `frc_eia923` is aggregating on correct columns (has no natural PK)

### CODE REVIEW

**Comment by TrentonBush** [comment]
For the release notes, I might mention that EPA CEMS was already updated to 2020. That is clear from the 0.4 release notes, but it is the only dataset *not* mentioned in 0.5 and it might confuse someone looking for it.

File: docs/release_notes.rst [hunk 1]
```diff
@@ -37,9 +52,77 @@ modular metadata management system that uses `Pydantic
 validate the metadata schema and export to a variety of formats to support data
```
Comment (cmgosnell) [inline docs/release_notes.rst:52]: github won't let me comment above this line... grr. but this is actually for the bullet above that starts with 

> * The new :mod:`pudl.load.sqlite` and :mod:`pudl.load.parquet` modules contain

The end of this bullet seem out of date bc #1211 is DONE.

File: docs/release_notes.rst [hunk 2]
```diff
@@ -37,9 +52,77 @@ modular metadata management system that uses `Pydantic
...
+* Many new foreign key constraints are being enforced between the EIA data tables,
+  entity tables, and coding tables. See :issue:`1196`.
+* Fuel types and energy sources reported to EIA are now defined in / constrained by
+  the static :ref:`energy_sources_eia` table.
+* The columns that indicate the mode of transport for various fuels now contain short
+  codes rather than longer labels, and are defined in / constrained by the static
+  :ref:`fuel_transportation_modes_eia` table.
+* In the simplified FERC 1 fuel type categories, we're now using `other` instead of
+  `unknown`.
+* Several columns have been renamed to harmonize meanings between different tables and
```
Comment (cmgosnell) [inline docs/release_notes.rst:96]: was there a corresponding issue for this?

**Review by cmgosnell — CHANGES_REQUESTED** [review]
bb suggestion in release notes.

File: docs/release_notes.rst (hunk 2 above)
Comment (zaneselvans) [reply docs/release_notes.rst:96]: No, it was just something that came up in the general coding work. We had used "other" in the EIA data and "unknown" in the ferc1 data.

File: docs/release_notes.rst (hunk 1 above)
Comment (zaneselvans) [reply docs/release_notes.rst:52]: Okay, updated!

**Comment by zaneselvans** [comment]
Okay @TrentonBush I added a clarifying note on the CEMS data.

**Comment by zaneselvans** [comment]
Regarding aggregation of the `generation_fuel_eia923`, `generation_fuel_nuclear_eia923` and `boiler_fuel_eia923` tables to monthly and annual levels, see #1344

**Pull request merged**